src/pymor/models/iosys.py

```diff
@@ -917,6 +919,153 @@ def hankel_norm(self, mu=None):
+            return self.h2_norm()
+
+        K = bernoulli_stabilize(A, E, C.as_source_array(mu=mu), ast_spectrum, trans=False)
+        KC = LowRankOperator(K, np.eye(len(K)), C.as_source_array(mu=mu))
+
+        if not isinstance(D, ZeroOperator):
+            BmKD = B - LowRankOperator(K, np.eye(len(K)), D.as_source_array(mu=mu))
+        else:
+            BmKD = B
+
+        if self.dim_input <= self.dim_output:
+            cf = solve_lyap_lrcf(A - KC, E, BmKD.as_range_array(mu=mu),
+                                 trans=False, options=options_lrcf)
+            return np.sqrt(self.C.apply(cf, mu=mu).norm2().sum())
+        else:
+            of = solve_lyap_lrcf(A - KC, E, C.as_source_array(mu=mu),
+                                 trans=True, options=options_lrcf)
+            return np.sqrt(BmKD.apply_adjoint(of, mu=mu).norm2().sum())
+
+    @cached
+    def linf_norm(self, mu=None, return_fpeak=False, ab13dd_equilibrate=False):
+        """Compute the L_infinity-norm of the |LTIModel|.
+
+        Parameters
+        ----------
+        mu
+            |Parameter|.
+        return_fpeak
+            Whether to return the frequency at which the maximum is achieved.
+        ab13dd_equilibrate
+            Whether `slycot.ab13dd` should use equilibration.
+
+        Returns
+        -------
+        norm
+            L_infinity-norm.
+        fpeak
+            Frequency at which the maximum is achieved (if `return_fpeak` is `True`).
+        """
+        return self.hinf_norm(mu=mu, return_fpeak=return_fpeak, ab13dd_equilibrate=ab13dd_equilibrate)
+
+    def get_ast_spectrum(self, ast_pole_data=None, mu=None):
+        """Compute anti-stable subset of the poles of the |LTIModel|.
+
+        Parameters
+        ----------
+        ast_pole_data
+            Can be:
+
+            - dictionary of parameters for :func:`~pymor.algorithms.eigs.eigs`,
+            - list of anti-stable eigenvalues (scalars),
+            - tuple `(lev, ew, rev)` where `ew` contains the anti-stable eigenvalues
+              and `lev` and `rev` are |VectorArrays| representing the eigenvectors.
+            - `None` if anti-stable eigenvalues should be computed via dense methods.
+        mu
+            |Parameter|.
+
+        Returns
+        -------
+        lev
+            |VectorArray| of left eigenvectors.
+        ew
+            One-dimensional |NumPy array| of anti-stable eigenvalues.
+        rev
+            |VectorArray| of right eigenvectors.
+        """
+        A, B, C, D, E = (op.assemble(mu=mu) for op in [self.A, self.B, self.C, self.D, self.E])
+
+        if ast_pole_data is not None:
+            if type(ast_pole_data) == dict:
+                ew, rev = eigs(A, E=E if self.E else None, left_evp=False, **ast_pole_data)
+                ast_idx = np.where(ew.real > 0.)
+                ast_ews = ew[ast_idx]
+                if len(ast_ews) == 0:
+                    return (self.solution_space.empty(), [], self.solution_space.empty())
```

Comment:
For a consistent return type:

```suggestion
                    return (self.solution_space.empty(), np.empty((0,)), self.solution_space.empty())
```